packages/swing-store/src/archiver.js

```diff
@@ -0,0 +1,44 @@
+import { finished as streamFinishedCallback, Readable } from 'node:stream';
+import { promisify } from 'node:util';
+import { createGzip } from 'node:zlib';
+import { withDeferredCleanup } from '@agoric/internal/src/node/utils.js';
+
+const streamFinished = promisify(streamFinishedCallback);
+
+/*
+ * @param {string} dirPath
+ * @param {object} powers
+ * @param {Pick<import('fs'), 'createWriteStream' | 'mkdirSync' | 'renameSync'>} powers.fs
+ * @param {Pick<import('path'), 'join'>} powers.path
+ * @param {Pick<import('tmp'), 'fileSync'>} powers.tmp
+ */
+export const makeArchiveTranscript = (dirPath, powers) => {
+  const { fs, path, tmp } = powers;
+  fs.mkdirSync(dirPath, { recursive: true });
+  const archiveTranscript = (spanName, entries) => {
+    const destPath = path.join(dirPath, `${spanName}.gz`);
+    return withDeferredCleanup(async addCleanup => {
+      const {
+        name: tmpName,
+        fd,
+        removeCallback,
+      } = tmp.fileSync({
+        prefix: spanName,
+        postfix: '.gz',
+        detachDescriptor: true,
```

Comment:
why do we care about detaching the descriptor?

Reply:
The library complained without it; because of `streamFinished(writer)` IIRC.

Reply:
Oh right, stream will close the fd unless you tell it not to.